main/src/main/scala/sbt/Keys.scala

```diff
@@ -59,6 +59,7 @@ object Keys {
   val logBuffered = settingKey[Boolean]("True if logging should be buffered until work completes.").withRank(CSetting)
   val sLog = settingKey[Logger]("Logger usable by settings during project loading.").withRank(CSetting)
   val serverLog = taskKey[Unit]("A dummy task to set server log level using Global / serverLog / logLevel.").withRank(CTask)
+  val showFullStackTrace = settingKey[Boolean]("Display the full stack trace including internal sbt classes if a task fails").withRank(DSetting)
```

Comment:
See https://github.com/sbt/sbt/issues/3793

Reply:
I think these are separate issues. The trace level sets a fixed limit on the number of lines of the stack trace to display. That being said, I think it would be straightforward to also apply the trace level as well in the portion of EvaluateTask where I truncate the stack trace. It might not fully resolve #3793 but would be a step in the right direction.

Reply:
https://github.com/sbt/util/blob/d4824a53ae23afffc6178d77032f27b229f414c4/internal/util-logging/src/main/scala/sbt/internal/util/StackTrace.scala#L18-L22

"If d is 0, then all elements are included up to (but not including) the first element that comes from sbt." so traceLevel set to `0` is the "cleaned up" version, and traceLevel set to `Int.MaxValue` is the raw version, roughly mapping to `false` and `true` in your new setting?

Reply:
ok, I think I have the fix - https://github.com/sbt/util/pull/211

Reply:
Ok. Are you going to reopen #4963? I like the change to set trace level to 0 on interactive sessions. The only very minor issue I have with the logger approach is that my approach also strips `[error] at scala.Function1.$anonfun$compose$1(Function1.scala:49)`. Whenever a task depends on another task, a line like that will appear. Your test task didn't depend on any other tasks so its stack trace was slightly simpler.

Reply:
I'll open another PR to bump util. The current default is already `-1` so that doesn't need to change. I'll see how it looks like with more complicated task.